Sources/AWSSDKSwiftCore/AWSClient.swift

```diff
@@ -334,6 +334,9 @@ extension AWSClient {
             case .restjson:
```

Comment:
I, personally, don't like switches within switches? Maybe this is a chance for:

```swift
switch (awsRequest.httpMethod, self.serviceProtocol.type, awsRequest.httpHeaders.count) {
case ("GET",  .restjson, _), 
     ("HEAD", .restjson, _):
    return try createNIORequestWithSignedHeader(awsRequest)
case ("GET",  _, 0),
     ("HEAD", _, 0):
    return try createNIORequestWithSignedURL(awsRequest)
case ("GET",  _, 1...),
     ("HEAD", _, 1...):
    return try createNIORequestWithSignedHeader(awsRequest)
}
```

Reply:
Not totally keen on that. Combining the httpMethod and type I'm ok with but including the header count just seems wrong. 
```swift
switch (awsRequest.httpMethod, self.serviceProtocol.type) {
case ("GET",  .restjson), ("HEAD", .restjson):
    return try createNIORequestWithSignedHeader(awsRequest)
case ("GET",  _), ("HEAD", _):
    if awsRequest.httpHeaders.count > 0 {
        return try createNIORequestWithSignedHeader(awsRequest)
    }
    return try createNIORequestWithSignedURL(awsRequest)
default:
    return try createNIORequestWithSignedHeader(awsRequest)
}
```
Also will that mean the switch tests for the header count range twice?

Reply:
I can live with that totally fine!